### PR TITLE
Add collection filters block

### LINKED ISSUE #10711: Create Collection Filters Block
Ref design discussion https://github.com/woocommerce/woocommerce-blocks/discussions/10691

This issue is to create a new Collection Filters block that will eventually house the filter blocks we currently have (price,rating,attributes and stock).

### PR DESCRIPTION
**Pull request by roykho**
## What

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/10711

This PR adds a new block called `Collection Filters` block. It will serve as a wrapper container for all active filters we have. It is also a dependent of the `Product Collection` block.

At this point, the block is just a simple block act as the wrapper for other blocks, we want to merge this PR to trunk to make develop other inner block easier. This PR won't close its connected issue as we haven't fully implemented the block yet.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. On a post/page add the `Product Collection` block.
2. Using the block inserter within the `Product Collection` block, try to add the `Collection Filters` block. Note that it will not work outside of the `Product Collection` Block.
3. You should now be able to select the filter you want to use.
4. Randomly choose one and ensure it indeed adds the filter to the page and save.
5. Go to the frontend and ensure you see the filter you've added within the `Product Collection` block.

* [x] Do not include in the Testing Notes 
* [x] Should be tested by the development team exclusively 

## Screenshots or screencast

| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental
* [ ] N/A

## Checklist

Required:
* [ ] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Add suggested changelog entry here.

### CODE REVIEW

File: assets/js/blocks/collection-filters/edit.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	return (
+		<div { ...blockProps }>
+			<InnerBlocks
+				allowedBlocks={ [
+					'woocommerce/attribute-filter',
+					'woocommerce/price-filter',
+					'woocommerce/rating-filter',
+					'woocommerce/stock-filter',
+					'woocommerce/filter-wrapper',
+				] }
```
Comment (dinhtungdu) [inline assets/js/blocks/collection-filters/edit.tsx]: I think we should remove this, at least during the development.

**Review by dinhtungdu — COMMENTED** [review]
@roykho @sunyatasattva Can we merge this one to trunk to make working on inner blocks easier.

File: assets/js/blocks/collection-filters/edit.tsx (hunk 1 above)
Comment (roykho) [reply assets/js/blocks/collection-filters/edit.tsx]: @dinhtungdu - sure go ahead and do that.

File: assets/js/blocks/collection-filters/save.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/**
+ * External dependencies
+ */
+import { useBlockProps, useInnerBlocksProps } from '@wordpress/block-editor';
+
+export default function save() {
+	const blockProps = useBlockProps.save();
+	const innerBlockProps = useInnerBlocksProps.save( blockProps );
+
+	return <div { ...innerBlockProps } />;
```
Comment (sunyatasattva) [inline assets/js/blocks/collection-filters/save.tsx]: My usual nitpicks… what about using a different tag? Personally, I find `nav` pretty apt as the container of various filters. To the very least, I would say `section` is appropriate.

**Review by sunyatasattva — APPROVED** [review]
Usual Lucio style nitpicks, but am approving!

File: assets/js/blocks/collection-filters/save.tsx (hunk 2 above)
Comment (dinhtungdu) [reply assets/js/blocks/collection-filters/save.tsx]: Agree, let's use `<nav>` for now.

@sunyatasattva I think merchants may have different preferences on the tags they want to use, should we add an option here to let merchants control which tag to use?

Comment (dinhtungdu) [reply assets/js/blocks/collection-filters/save.tsx]: Moved this to the parent issue. https://github.com/woocommerce/woocommerce-blocks/issues/10711#issuecomment-1718771574

**Pull request merged**